### ISSUE TITLE
Memory increasing with v0.1 SCREAM cases using GNU 9 

### ISSUE THREAD

**Issue by ndkeen** [E3SM-Project/scream#1318]
Trying ne30 F (and ne4 F case) case on perlmutter, which uses GNU v9, I clearly see memory increasing over time.
`--res ne30pg2_ne30pg2 --mach perlmutter --compiler gnu --compset F2010-SCREAM-LR`

```
 memory_write: model date =   00010101       0 memory =    1296.07 MB (highwater)        415.80 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010102       0 memory =    1646.13 MB (highwater)        658.85 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010103       0 memory =    1935.73 MB (highwater)        894.62 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010104       0 memory =    2226.88 MB (highwater)       1119.89 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010105       0 memory =    2518.55 MB (highwater)       1351.52 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010106       0 memory =    2791.70 MB (highwater)       1576.65 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010107       0 memory =    3064.14 MB (highwater)       1807.30 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010108       0 memory =    3331.05 MB (highwater)       2032.07 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010109       0 memory =    3604.51 MB (highwater)       2263.69 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010110       0 memory =    3896.95 MB (highwater)       2496.23 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010111       0 memory =    4189.39 MB (highwater)       2728.54 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010112       0 memory =    4463.83 MB (highwater)       2960.90 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010113       0 memory =    4738.24 MB (highwater)       3187.36 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
 memory_write: model date =   00010114       0 memory =    5027.97 MB (highwater)       3412.55 MB (usage)  (pe=    0 comps= cpl ATM LND ICE OCN GLC ROF WAV IAC ESP)
```

I also verified that these tests: 
```
SMS_D_P64x1_Lm1.ne4pg2_ne4pg2.F2010-SCREAM-LR
SMS_P64x1_Lm1.ne4pg2_ne4pg2.F2010-SCREAM-LR
```
also show same memory increase issue on PM -- actually shoots up fairly rapidly.

**Comment by bartgol** [comment]
@brhillman Could this be somehow related to what you observed for SCREAMv1 on Summit? If so, could the common root cause be in rrtmgp?

**Comment by ndkeen** [comment]
Well this is similar behavior to what I found with MMF (using gnu9 not gnu8) that we narrowed down to being in the F90 radiation interface.  Is that what you mean?

**Comment by bartgol** [comment]
Maybe? I'm not sure. I don't recall if you were on the v1 telecon today, but in short, Ben noticed that the memory pool allocator of YAKL keeps allocating memory when running SCREAMv1 on Summit, eventually causing a crash. I think he said this doesn't happen for e3sm, so I assumed we were doing something wrong inside _our_ interface to radiation. But if you say this happens also in MMF, and in v0, then it might be a problem upstream. But I'll let Ben explain this better.

**Comment by ndkeen** [comment]
Ah. I was on the telecon and do recall Ben mentioned this, but I thought he was running the v1 SCREAM, which I'm not. Though there could still be something in common here.  With MMF at least, we found that the memory did not increase with the new rrtgmpxx.

**Comment by bartgol** [comment]
Ah, ok, then it might be an unrelated issue. Yes, I think Ben was running v1 SCREAM. But since the issue is somewhat similar, there might be a common connection.

**Comment by brhillman** [comment]
Shoot sorry I missed this. To clarify, the issue Noel was seeing with MMF was with the F90 RRTMGP code. The issue went away when we switched to the C++ code in MMF, so we figured there was a bug in the F90 code (or rather a compiler bug only exposed by the F90 code) rather than the interface itself. This issue Noel is seeing is with v0.x, which is also currently using the F90 code in common with EAM, so I think this memory leak is likely associated with the F90 code and will maybe go away once we do an upstream merge and can switch to the C++ code. I suspect the memory leak in V1 IS our (my) fault though, since that is already using the C++ code, and I haven't seen this behavior in other contexts yet.
